### PR TITLE
Fix lgtm issue

### PR DESCRIPTION
**Pull request by TerryFei**
Fix below LGTM issue.
HttpClient created with CheckCertificateRevocationList disabled
Unsafe TypeNameHandling

### CODE REVIEW

File: test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentBuilderTest.cs [hunk 1]
```diff
@@ -994,8 +994,8 @@ public async Task TestBuildWithXrefService()
                 StatusCode = HttpStatusCode.OK,
                 Content = new StringContent("[{'uid':'csharp_coding_standards', 'name':'C# Coding Standards', 'href':'http://dotnet.github.io/docfx/guideline/csharp_coding_standards.html'}]")
             });
-
-            var httpClient = new HttpClient(fakeResponseHandler);
+            // this method only used for unit test.
```
Comment (live1206) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentBuilderTest.cs:997]: remove this comment since this is a test class already

Comment (TerryFei) [reply test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentBuilderTest.cs]: done

**Review by live1206 — APPROVED** [review]

**Review by herohua — APPROVED** [review]

File: test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentProcessors/YamlDocumentProcessor.cs [hunk 2]
```diff
@@ -175,7 +175,7 @@ protected virtual JsonSerializer GetSerializer()
                 {
                     new Newtonsoft.Json.Converters.StringEnumConverter(),
                 },
-                TypeNameHandling = TypeNameHandling.All,
+                TypeNameHandling = TypeNameHandling.All, // lgtm [cs/unsafe-type-name-handling]
```
Comment (KalleOlaviNiemitalo) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentProcessors/YamlDocumentProcessor.cs:178]: Should this have a comment about why it's safe? Like, is the input data part of this repository and not controllable by adversaries?

Comment (TerryFei) [reply test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentProcessors/YamlDocumentProcessor.cs:178]: this is a test class. input data is test data.

**Comment by herohua** [comment]
just fixed gated CI, trigger by reopening the PR

**Pull request merged**